internal/store/device.go

```diff
@@ -107,9 +110,13 @@ func (s *DeviceStore) Create(ctx context.Context, orgId uuid.UUID, resource *api
 	if resource == nil {
 		return nil, flterrors.ErrResourceIsNil
 	}
-	device := model.NewDeviceFromApiResource(resource)
+	device, err := model.NewDeviceFromApiResource(resource)
+	if err != nil {
+		return nil, err
```

Comment:
flterrors.ErrIllegalResourceVersionFormat ?

Reply:
It is created here https://github.com/flightctl/flightctl/blob/2c89591e575ea3e52de6a46878404b0c940e4cd2/internal/store/model/device.go#L77
This way the function may return other errors (in the future)